frontend/packages/dev-console/src/components/topology/TopologyPage.tsx

```diff
@@ -1,136 +1,133 @@
+    },
+    [activeViewStorageKey],
+  );
 
   const handleNamespaceChange = (ns: string) => {
     if (ns !== namespace) {
       removeQueryArgument(TOPOLOGY_SEARCH_FILTER_KEY);
     }
   };
 
-  React.useEffect(() => setTopologyActiveView(showListView && !showGraphView ? 'list' : 'graph'), [
-    showListView,
-    showGraphView,
-  ]);
-
-  if (!showGraphView && !showListView) {
-    return (
-      <Redirect
-        to={`/topology/${namespace ? `ns/${namespace}` : 'all-namespaces'}/${
-          getTopologyActiveView() === 'list' ? 'list' : 'graph'
-        }${queryParams ? `?${queryParams.toString()}` : ''}`}
-      />
-    );
-  }
-
   return (
-    <>
+    <DataModelProvider namespace={namespace}>
       <Helmet>
         <title>Topology</title>
       </Helmet>
-      <NamespacedPage
-        variant={showListView ? NamespacedPageVariants.light : NamespacedPageVariants.default}
-        onNamespaceChange={handleNamespaceChange}
-        toolbar={
-          namespace && !dataModelContext.isEmptyModel ? (
-            <>
-              {!showListView && namespace && (
-                <Popover
-                  aria-label="Shortcuts"
-                  bodyContent={TopologyShortcuts}
-                  position="left"
-                  maxWidth="100vw"
-                >
-                  <Button
-                    type="button"
-                    variant="link"
-                    className="odc-topology__shortcuts-button"
-                    icon={<QuestionCircleIcon />}
-                    data-test-id="topology-view-shortcuts"
-                  >
-                    View shortcuts
-                  </Button>
-                </Popover>
-              )}
-              <Tooltip position="left" content={showListView ? 'Topology View' : 'List View'}>
-                <Link
-                  className="pf-c-button pf-m-plain odc-topology__view-switcher"
-                  to={`/topology/${namespace ? `ns/${namespace}` : 'all-namespaces'}${
-                    showListView ? '/graph' : '/list'
-                  }${queryParams ? `?${queryParams.toString()}` : ''}`}
-                >
-                  {showListView ? <TopologyIcon size="md" /> : <ListIcon size="md" />}
-                </Link>
-              </Tooltip>
-            </>
-          ) : null
-        }
-      >
-        <Firehose resources={[{ kind: 'Project', prop: 'projects', isList: true }]}>
-          <ProjectsExistWrapper title="Topology">
-            {namespace ? (
-              <TopologyDataRenderer showGraphView={showGraphView} />
-            ) : (
-              <CreateProjectListPage title="Topology">
-                Select a project to view the topology
-              </CreateProjectListPage>
-            )}
-          </ProjectsExistWrapper>
-        </Firehose>
-      </NamespacedPage>
-    </>
+      <div className="odc-topology-page">
+        <NamespaceBar onNamespaceChange={handleNamespaceChange} hideProjects={hideProjects}>
```

Comment:
Why the move away from `NamespacedPage`?

Reply:
Changed it back. I had tried going in another direction that is no longer relevant.